--- original
+++ modified
@@ -2974,7 +2974,11 @@
 }
 pub const HW_BREAKPOINT_LEN_1: _bindgen_ty_1 = 1;
 pub const HW_BREAKPOINT_LEN_2: _bindgen_ty_1 = 2;
+pub const HW_BREAKPOINT_LEN_3: _bindgen_ty_1 = 3;
 pub const HW_BREAKPOINT_LEN_4: _bindgen_ty_1 = 4;
+pub const HW_BREAKPOINT_LEN_5: _bindgen_ty_1 = 5;
+pub const HW_BREAKPOINT_LEN_6: _bindgen_ty_1 = 6;
+pub const HW_BREAKPOINT_LEN_7: _bindgen_ty_1 = 7;
 pub const HW_BREAKPOINT_LEN_8: _bindgen_ty_1 = 8;
 pub type _bindgen_ty_1 = u32;
 pub const HW_BREAKPOINT_EMPTY: _bindgen_ty_2 = 0;
