--- original
+++ modified
@@ -528,6 +528,7 @@
 pub const PERF_MAX_CONTEXTS_PER_STACK: u32 = 8;
 pub const PERF_AUX_FLAG_TRUNCATED: u32 = 1;
 pub const PERF_AUX_FLAG_OVERWRITE: u32 = 2;
+pub const PERF_AUX_FLAG_PARTIAL: u32 = 4;
 pub const PERF_FLAG_FD_NO_GROUP: u32 = 1;
 pub const PERF_FLAG_FD_OUTPUT: u32 = 2;
 pub const PERF_FLAG_PID_CGROUP: u32 = 4;
@@ -1823,8 +1824,43 @@
     }
     #[inline]
     #[allow(unnecessary_transmutes)]
+    pub fn namespaces(&self) -> __u64 {
+        unsafe { ::std::mem::transmute(self._bitfield_1.get_const::<28usize, 1u8>() as u64) }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
+    pub fn set_namespaces(&mut self, val: __u64) {
+        unsafe {
+            let val: u64 = ::std::mem::transmute(val);
+            self._bitfield_1.set_const::<28usize, 1u8>(val as u64)
+        }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
+    pub unsafe fn namespaces_raw(this: *const Self) -> __u64 {
+        unsafe {
+            ::std::mem::transmute(
+                <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_get_const::<28usize, 1u8>(
+                    ::std::ptr::addr_of!((*this)._bitfield_1),
+                ) as u64,
+            )
+        }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
+    pub unsafe fn set_namespaces_raw(this: *mut Self, val: __u64) {
+        unsafe {
+            let val: u64 = ::std::mem::transmute(val);
+            <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_set_const::<28usize, 1u8>(
+                ::std::ptr::addr_of_mut!((*this)._bitfield_1),
+                val as u64,
+            )
+        }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
     pub fn __reserved_1(&self) -> __u64 {
-        unsafe { ::std::mem::transmute(self._bitfield_1.get_const::<28usize, 36u8>() as u64) }
+        unsafe { ::std::mem::transmute(self._bitfield_1.get_const::<29usize, 35u8>() as u64) }
     }
     #[inline]
     #[allow(unnecessary_transmutes)]
@@ -1831,7 +1867,7 @@
     pub fn set___reserved_1(&mut self, val: __u64) {
         unsafe {
             let val: u64 = ::std::mem::transmute(val);
-            self._bitfield_1.set_const::<28usize, 36u8>(val as u64)
+            self._bitfield_1.set_const::<29usize, 35u8>(val as u64)
         }
     }
     #[inline]
@@ -1839,8 +1875,8 @@
     pub unsafe fn __reserved_1_raw(this: *const Self) -> __u64 {
         unsafe {
             ::std::mem::transmute(<__BindgenBitfieldUnit<[u8; 8usize]>>::raw_get_const::<
-                28usize,
-                36u8,
+                29usize,
+                35u8,
             >(::std::ptr::addr_of!((*this)._bitfield_1))
                 as u64)
         }
@@ -1850,7 +1886,7 @@
     pub unsafe fn set___reserved_1_raw(this: *mut Self, val: __u64) {
         unsafe {
             let val: u64 = ::std::mem::transmute(val);
-            <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_set_const::<28usize, 36u8>(
+            <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_set_const::<29usize, 35u8>(
                 ::std::ptr::addr_of_mut!((*this)._bitfield_1),
                 val as u64,
             )
@@ -1886,6 +1922,7 @@
         use_clockid: __u64,
         context_switch: __u64,
         write_backward: __u64,
+        namespaces: __u64,
         __reserved_1: __u64,
     ) -> __BindgenBitfieldUnit<[u8; 8usize]> {
         let mut __bindgen_bitfield_unit: __BindgenBitfieldUnit<[u8; 8usize]> = Default::default();
@@ -1999,7 +2036,11 @@
             let write_backward: u64 = unsafe { ::std::mem::transmute(write_backward) };
             write_backward as u64
         });
-        __bindgen_bitfield_unit.set_const::<28usize, 36u8>({
+        __bindgen_bitfield_unit.set_const::<28usize, 1u8>({
+            let namespaces: u64 = unsafe { ::std::mem::transmute(namespaces) };
+            namespaces as u64
+        });
+        __bindgen_bitfield_unit.set_const::<29usize, 35u8>({
             let __reserved_1: u64 = unsafe { ::std::mem::transmute(__reserved_1) };
             __reserved_1 as u64
         });
@@ -2398,6 +2439,30 @@
     ["Offset of field: perf_event_header::size"]
         [::std::mem::offset_of!(perf_event_header, size) - 6usize];
 };
+#[repr(C)]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct perf_ns_link_info {
+    pub dev: __u64,
+    pub ino: __u64,
+}
+#[allow(clippy::unnecessary_operation, clippy::identity_op)]
+const _: () = {
+    ["Size of perf_ns_link_info"][::std::mem::size_of::<perf_ns_link_info>() - 16usize];
+    ["Alignment of perf_ns_link_info"][::std::mem::align_of::<perf_ns_link_info>() - 8usize];
+    ["Offset of field: perf_ns_link_info::dev"]
+        [::std::mem::offset_of!(perf_ns_link_info, dev) - 0usize];
+    ["Offset of field: perf_ns_link_info::ino"]
+        [::std::mem::offset_of!(perf_ns_link_info, ino) - 8usize];
+};
+pub const NET_NS_INDEX: _bindgen_ty_4 = 0;
+pub const UTS_NS_INDEX: _bindgen_ty_4 = 1;
+pub const IPC_NS_INDEX: _bindgen_ty_4 = 2;
+pub const PID_NS_INDEX: _bindgen_ty_4 = 3;
+pub const USER_NS_INDEX: _bindgen_ty_4 = 4;
+pub const MNT_NS_INDEX: _bindgen_ty_4 = 5;
+pub const CGROUP_NS_INDEX: _bindgen_ty_4 = 6;
+pub const NR_NAMESPACES: _bindgen_ty_4 = 7;
+pub type _bindgen_ty_4 = u32;
 pub const PERF_RECORD_MMAP: perf_event_type = 1;
 pub const PERF_RECORD_LOST: perf_event_type = 2;
 pub const PERF_RECORD_COMM: perf_event_type = 3;
@@ -2413,7 +2478,8 @@
 pub const PERF_RECORD_LOST_SAMPLES: perf_event_type = 13;
 pub const PERF_RECORD_SWITCH: perf_event_type = 14;
 pub const PERF_RECORD_SWITCH_CPU_WIDE: perf_event_type = 15;
-pub const PERF_RECORD_MAX: perf_event_type = 16;
+pub const PERF_RECORD_NAMESPACES: perf_event_type = 16;
+pub const PERF_RECORD_MAX: perf_event_type = 17;
 pub type perf_event_type = u32;
 pub const PERF_CONTEXT_HV: perf_callchain_context = 18446744073709551584;
 pub const PERF_CONTEXT_KERNEL: perf_callchain_context = 18446744073709551488;
