--- original
+++ modified
@@ -608,6 +608,7 @@
 pub const PERF_MEM_LVLNUM_L4: u32 = 4;
 pub const PERF_MEM_LVLNUM_L2_MHB: u32 = 5;
 pub const PERF_MEM_LVLNUM_MSC: u32 = 6;
+pub const PERF_MEM_LVLNUM_L0: u32 = 7;
 pub const PERF_MEM_LVLNUM_UNC: u32 = 8;
 pub const PERF_MEM_LVLNUM_CXL: u32 = 9;
 pub const PERF_MEM_LVLNUM_IO: u32 = 10;
@@ -646,6 +647,23 @@
 pub const PERF_MEM_HOPS_2: u32 = 3;
 pub const PERF_MEM_HOPS_3: u32 = 4;
 pub const PERF_MEM_HOPS_SHIFT: u32 = 43;
+pub const PERF_MEM_REGION_NA: u32 = 0;
+pub const PERF_MEM_REGION_RSVD: u32 = 1;
+pub const PERF_MEM_REGION_L_SHARE: u32 = 2;
+pub const PERF_MEM_REGION_L_NON_SHARE: u32 = 3;
+pub const PERF_MEM_REGION_O_IO: u32 = 4;
+pub const PERF_MEM_REGION_O_SHARE: u32 = 5;
+pub const PERF_MEM_REGION_O_NON_SHARE: u32 = 6;
+pub const PERF_MEM_REGION_MMIO: u32 = 7;
+pub const PERF_MEM_REGION_MEM0: u32 = 8;
+pub const PERF_MEM_REGION_MEM1: u32 = 9;
+pub const PERF_MEM_REGION_MEM2: u32 = 10;
+pub const PERF_MEM_REGION_MEM3: u32 = 11;
+pub const PERF_MEM_REGION_MEM4: u32 = 12;
+pub const PERF_MEM_REGION_MEM5: u32 = 13;
+pub const PERF_MEM_REGION_MEM6: u32 = 14;
+pub const PERF_MEM_REGION_MEM7: u32 = 15;
+pub const PERF_MEM_REGION_SHIFT: u32 = 46;
 pub const PERF_BRANCH_ENTRY_INFO_BITS_MAX: u32 = 33;
 pub const BPF_TAG_SIZE: u32 = 8;
 pub type __u8 = ::std::os::raw::c_uchar;
@@ -3748,8 +3766,43 @@
     }
     #[inline]
     #[allow(unnecessary_transmutes)]
+    pub fn mem_region(&self) -> __u64 {
+        unsafe { ::std::mem::transmute(self._bitfield_1.get_const::<46usize, 5u8>() as u64) }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
+    pub fn set_mem_region(&mut self, val: __u64) {
+        unsafe {
+            let val: u64 = ::std::mem::transmute(val);
+            self._bitfield_1.set_const::<46usize, 5u8>(val as u64)
+        }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
+    pub unsafe fn mem_region_raw(this: *const Self) -> __u64 {
+        unsafe {
+            ::std::mem::transmute(
+                <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_get_const::<46usize, 5u8>(
+                    ::std::ptr::addr_of!((*this)._bitfield_1),
+                ) as u64,
+            )
+        }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
+    pub unsafe fn set_mem_region_raw(this: *mut Self, val: __u64) {
+        unsafe {
+            let val: u64 = ::std::mem::transmute(val);
+            <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_set_const::<46usize, 5u8>(
+                ::std::ptr::addr_of_mut!((*this)._bitfield_1),
+                val as u64,
+            )
+        }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
     pub fn mem_rsvd(&self) -> __u64 {
-        unsafe { ::std::mem::transmute(self._bitfield_1.get_const::<46usize, 18u8>() as u64) }
+        unsafe { ::std::mem::transmute(self._bitfield_1.get_const::<51usize, 13u8>() as u64) }
     }
     #[inline]
     #[allow(unnecessary_transmutes)]
@@ -3756,7 +3809,7 @@
     pub fn set_mem_rsvd(&mut self, val: __u64) {
         unsafe {
             let val: u64 = ::std::mem::transmute(val);
-            self._bitfield_1.set_const::<46usize, 18u8>(val as u64)
+            self._bitfield_1.set_const::<51usize, 13u8>(val as u64)
         }
     }
     #[inline]
@@ -3764,8 +3817,8 @@
     pub unsafe fn mem_rsvd_raw(this: *const Self) -> __u64 {
         unsafe {
             ::std::mem::transmute(<__BindgenBitfieldUnit<[u8; 8usize]>>::raw_get_const::<
-                46usize,
-                18u8,
+                51usize,
+                13u8,
             >(::std::ptr::addr_of!((*this)._bitfield_1))
                 as u64)
         }
@@ -3775,7 +3828,7 @@
     pub unsafe fn set_mem_rsvd_raw(this: *mut Self, val: __u64) {
         unsafe {
             let val: u64 = ::std::mem::transmute(val);
-            <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_set_const::<46usize, 18u8>(
+            <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_set_const::<51usize, 13u8>(
                 ::std::ptr::addr_of_mut!((*this)._bitfield_1),
                 val as u64,
             )
@@ -3794,6 +3847,7 @@
         mem_snoopx: __u64,
         mem_blk: __u64,
         mem_hops: __u64,
+        mem_region: __u64,
         mem_rsvd: __u64,
     ) -> __BindgenBitfieldUnit<[u8; 8usize]> {
         let mut __bindgen_bitfield_unit: __BindgenBitfieldUnit<[u8; 8usize]> = Default::default();
@@ -3837,7 +3891,11 @@
             let mem_hops: u64 = unsafe { ::std::mem::transmute(mem_hops) };
             mem_hops as u64
         });
-        __bindgen_bitfield_unit.set_const::<46usize, 18u8>({
+        __bindgen_bitfield_unit.set_const::<46usize, 5u8>({
+            let mem_region: u64 = unsafe { ::std::mem::transmute(mem_region) };
+            mem_region as u64
+        });
+        __bindgen_bitfield_unit.set_const::<51usize, 13u8>({
             let mem_rsvd: u64 = unsafe { ::std::mem::transmute(mem_rsvd) };
             mem_rsvd as u64
         });
