--- original
+++ modified
@@ -525,6 +525,7 @@
 pub const PERF_RECORD_MISC_EXACT_IP: u32 = 16384;
 pub const PERF_RECORD_MISC_EXT_RESERVED: u32 = 32768;
 pub const PERF_MAX_STACK_DEPTH: u32 = 127;
+pub const PERF_MAX_CONTEXTS_PER_STACK: u32 = 8;
 pub const PERF_AUX_FLAG_TRUNCATED: u32 = 1;
 pub const PERF_AUX_FLAG_OVERWRITE: u32 = 2;
 pub const PERF_FLAG_FD_NO_GROUP: u32 = 1;
@@ -1784,8 +1785,43 @@
     }
     #[inline]
     #[allow(unnecessary_transmutes)]
+    pub fn write_backward(&self) -> __u64 {
+        unsafe { ::std::mem::transmute(self._bitfield_1.get_const::<27usize, 1u8>() as u64) }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
+    pub fn set_write_backward(&mut self, val: __u64) {
+        unsafe {
+            let val: u64 = ::std::mem::transmute(val);
+            self._bitfield_1.set_const::<27usize, 1u8>(val as u64)
+        }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
+    pub unsafe fn write_backward_raw(this: *const Self) -> __u64 {
+        unsafe {
+            ::std::mem::transmute(
+                <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_get_const::<27usize, 1u8>(
+                    ::std::ptr::addr_of!((*this)._bitfield_1),
+                ) as u64,
+            )
+        }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
+    pub unsafe fn set_write_backward_raw(this: *mut Self, val: __u64) {
+        unsafe {
+            let val: u64 = ::std::mem::transmute(val);
+            <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_set_const::<27usize, 1u8>(
+                ::std::ptr::addr_of_mut!((*this)._bitfield_1),
+                val as u64,
+            )
+        }
+    }
+    #[inline]
+    #[allow(unnecessary_transmutes)]
     pub fn __reserved_1(&self) -> __u64 {
-        unsafe { ::std::mem::transmute(self._bitfield_1.get_const::<27usize, 37u8>() as u64) }
+        unsafe { ::std::mem::transmute(self._bitfield_1.get_const::<28usize, 36u8>() as u64) }
     }
     #[inline]
     #[allow(unnecessary_transmutes)]
@@ -1792,7 +1828,7 @@
     pub fn set___reserved_1(&mut self, val: __u64) {
         unsafe {
             let val: u64 = ::std::mem::transmute(val);
-            self._bitfield_1.set_const::<27usize, 37u8>(val as u64)
+            self._bitfield_1.set_const::<28usize, 36u8>(val as u64)
         }
     }
     #[inline]
@@ -1800,8 +1836,8 @@
     pub unsafe fn __reserved_1_raw(this: *const Self) -> __u64 {
         unsafe {
             ::std::mem::transmute(<__BindgenBitfieldUnit<[u8; 8usize]>>::raw_get_const::<
-                27usize,
-                37u8,
+                28usize,
+                36u8,
             >(::std::ptr::addr_of!((*this)._bitfield_1))
                 as u64)
         }
@@ -1811,7 +1847,7 @@
     pub unsafe fn set___reserved_1_raw(this: *mut Self, val: __u64) {
         unsafe {
             let val: u64 = ::std::mem::transmute(val);
-            <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_set_const::<27usize, 37u8>(
+            <__BindgenBitfieldUnit<[u8; 8usize]>>::raw_set_const::<28usize, 36u8>(
                 ::std::ptr::addr_of_mut!((*this)._bitfield_1),
                 val as u64,
             )
@@ -1846,6 +1882,7 @@
         comm_exec: __u64,
         use_clockid: __u64,
         context_switch: __u64,
+        write_backward: __u64,
         __reserved_1: __u64,
     ) -> __BindgenBitfieldUnit<[u8; 8usize]> {
         let mut __bindgen_bitfield_unit: __BindgenBitfieldUnit<[u8; 8usize]> = Default::default();
@@ -1955,7 +1992,11 @@
             let context_switch: u64 = unsafe { ::std::mem::transmute(context_switch) };
             context_switch as u64
         });
-        __bindgen_bitfield_unit.set_const::<27usize, 37u8>({
+        __bindgen_bitfield_unit.set_const::<27usize, 1u8>({
+            let write_backward: u64 = unsafe { ::std::mem::transmute(write_backward) };
+            write_backward as u64
+        });
+        __bindgen_bitfield_unit.set_const::<28usize, 36u8>({
             let __reserved_1: u64 = unsafe { ::std::mem::transmute(__reserved_1) };
             __reserved_1 as u64
         });
@@ -2949,4 +2990,5 @@
 pub const PERF_IOC_OP_SET_FILTER: perf_ioc_ops = 1074275334;
 pub const PERF_IOC_OP_ID: perf_ioc_ops = 2148017159;
 pub const PERF_IOC_OP_SET_BPF: perf_ioc_ops = 1074013192;
+pub const PERF_IOC_OP_PAUSE_OUTPUT: perf_ioc_ops = 1074013193;
 pub type perf_ioc_ops = u32;
